### PR TITLE
David/recency2

### PR DESCRIPTION
**Pull request by David-Beglou**

### CODE REVIEW

**Review by MJHamar — CHANGES_REQUESTED** [review]
Nice work! Please add this to the constructor of the classes in `other_baselines.py` (with constructor call in the runner script and in the configs included)
Those are also affected by this parameter.

File: tgnnexplainer/xgraph/method/base_explainer_tg.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 class BaseExplainerTG(object):
     def __init__(self, model: Union[TGAN, None], model_name: str, explainer_name: str, dataset_name: str, all_events: str, explanation_level: str, device, 
-                verbose: bool = True, results_dir: Optional[str] = None, debug_mode: bool=True) -> None:
+                verbose: bool = True, results_dir: Optional[str] = None, debug_mode: bool=True, threshold_num=20) -> None:
```
Comment (m-krastev) [inline tgnnexplainer/xgraph/method/base_explainer_tg.py:14]: Small detail: is 25 more correct?

Comment (MJHamar) [reply tgnnexplainer/xgraph/method/base_explainer_tg.py:14]: Hard to say. Paper says 25 the code has 20 everywhere

File: benchmarks/xgraph/subgraphx_tg_run.py
Comment (MJHamar) [inline benchmarks/xgraph/subgraphx_tg_run.py:1]: not the most recent version

File: tgnnexplainer/xgraph/method/other_baselines_tg.py
Comment (MJHamar) [inline tgnnexplainer/xgraph/method/other_baselines_tg.py:1]: please don't override our work

File: tgnnexplainer/xgraph/method/other_baselines_tg.py [hunk 2]
```diff
@@ -314,19 +322,24 @@ def explain(self, node_idx=None, event_idx=None):
...
         for e_idx in e_idx_logit_dict.keys():
             # e_idx_score_dict[e_idx] = fidelity_inv_tg(ori_score, e_idx_logit_dict[e_idx]) # the larger the fidelity, the larger the unimportance.
             e_idx_score_dict[e_idx] = fidelity_inv_tg(ori_score, e_idx_logit_dict[e_idx]) * -1 # the larger the score, the larger the importance
-        
+
         # the same as Attn explainer
         candidate_weights = { e_idx: e_idx_score_dict[e_idx] for e_idx in self.candidate_events }
         candidate_weights = dict( sorted(candidate_weights.items(), key=lambda x: x[1], reverse=True) ) # NOTE: descending, important
-        
+
```
Comment (MJHamar) [inline tgnnexplainer/xgraph/method/other_baselines_tg.py:337]: why is this here? Candidates are selected in BaseExplainer

**Review by MJHamar — CHANGES_REQUESTED** [review]
Left some comments. Please be more careful of what you commit.
You also forgot about ATTNExplainer
